ironfish/src/rpc/server.ts

```diff
@@ -90,10 +91,11 @@ export class RpcServer {
     const rpcAuthToken = this.internal.get('rpcAuthToken')
 
     if (!rpcAuthToken) {
-      this.logger.debug(`Missing RPC Auth token in internal.json config.`)
       return false
     }
 
-    return requestAuthToken === rpcAuthToken
+    const userTokenBuffer = Buffer.from(requestAuthToken)
+    const authTokenBuffer = Buffer.from(rpcAuthToken)
+    return timingSafeEqual(userTokenBuffer, authTokenBuffer)
```

Comment:
Note that this will throw an error if the two buffers/tokens have a different length. The current version of this function would return `false` instead

Reply:
Good point. This didn't fail locally, so I'll fix the code and also add a test that will fail in this case.